src/avro/__tests__/AvroSchemaDecoder.spec.ts

```diff
@@ -0,0 +1,428 @@
+import {Writer} from '@jsonjoy.com/util/lib/buffers/Writer';
+import {AvroSchemaEncoder} from '../AvroSchemaEncoder';
+import {AvroSchemaDecoder} from '../AvroSchemaDecoder';
+import type {
+  AvroSchema,
+  AvroRecordSchema,
+  AvroEnumSchema,
+  AvroArraySchema,
+  AvroMapSchema,
+  AvroUnionSchema,
+  AvroFixedSchema,
+} from '../types';
+
+describe('AvroSchemaDecoder', () => {
+  let writer: Writer;
+  let encoder: AvroSchemaEncoder;
+  let decoder: AvroSchemaDecoder;
+
+  beforeEach(() => {
+    writer = new Writer();
+    encoder = new AvroSchemaEncoder(writer);
+    decoder = new AvroSchemaDecoder();
+  });
```

Comment:
Do not use `beforeEach`. To setup a testbed, create a helper function like `setup`:


```suggestion
  const setup = () => {
    writer = new Writer();
    encoder = new AvroSchemaEncoder(writer);
    decoder = new AvroSchemaDecoder();
    return {writer, encoder, decoder};
  };
```